### ISSUE TITLE
About the differences between predictions and queries.

### ISSUE THREAD

**Issue by fatfatfatmouse** [khdlr/HED-UNet#15]
1.   self.queries = nn.ModuleList([nn.Conv2d((1 << i) * bc, output_channels, 1)
                   for i in reversed(range(stack_height + 1))])
2.   self.predictors = nn.ModuleList([nn.Conv2d((1 << i) * bc, output_channels, 1)
                   for i in reversed(range(stack_height + 1))])

The queries and the predictors seem to be the same. 
So What are the differences between the results as follows？

1.full_scale_preds.append(F.interpolate(prediction, size=(H, W), mode='bilinear', align_corners=True))
2.queries = [F.interpolate(q(feat), size=(H, W), mode='bilinear', align_corners=True)
                       for q, feat in zip(self.queries, multilevel_features)]

**Comment by khdlr** [comment]
While the calculation is the same, the two modules have different functionality. The predictors are tasked with predicting the target masks/edges directly, while the queries are used for the attention merging mechanism to merge the predictors' outputs into a single, final prediction.

I recommend reading section III C) in the paper: https://ieeexplore.ieee.org/stamp/stamp.jsp?arnumber=9383809. Let me know if that clears your question or you have additional questions.

**Comment by fatfatfatmouse** [comment]
Thank you for taking the time to answer my question.I think my question has been completely answered.
I tried to use the resolution attention mechanism in your article to solve the problems in my research, and achieved good results.
Finally, thank you again for your meticulous and warm help.

**Comment by khdlr** [comment]
Glad to hear that! Feel free to reopen the issue if you have any more questions!
